sqlite/ext/comdb2/api_history.c

```diff
@@ -0,0 +1,94 @@
+/*
+   Copyright 2024 Bloomberg Finance L.P.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <stddef.h>
+
+#include "comdb2.h"
+#include "sqlite3.h"
+#include "ezsystables.h"
+
+extern struct dbenv *thedb;
+sqlite3_module systblApiHistoryModule = {
+    .access_flag = CDB2_ALLOW_ALL,
+    .systable_lock = "comdb2_tables",
+};
+
+typedef struct systable_api_history {
+    char *host;
+    char *task;
+    char *api_driver_name;
+    char *api_driver_version;
+    cdb2_client_datetime_t last_seen;
+} systable_api_history_t;
+
+int get_api_history_data(void **data, int *nrecords) {
+    acquire_api_history_lock(1); 
+    hash_t *raw_data = thedb->api_history;
+    *nrecords = hash_get_num_entries(raw_data);
+    systable_api_history_t *systable = calloc(*nrecords, sizeof(systable_api_history_t));
+
+    void *ent;
+    unsigned int bkt;
+    api_history_t *obj;
+
+    for (unsigned int i = 0; i < *nrecords; i++) {
+        obj = i ? (api_history_t *)hash_next(raw_data, &ent, &bkt) : 
+            (api_history_t *)hash_first(raw_data, &ent, &bkt);
+
+        systable[i].host = strdup(obj->host);
+        systable[i].task = strdup(obj->task);
+        systable[i].api_driver_name = strdup(obj->api_driver_name);
+        systable[i].api_driver_version = strdup(obj->api_driver_version);
+
+        cdb2_tm_t *time_info = (cdb2_tm_t *)localtime(&obj->last_seen);
+        systable[i].last_seen = (cdb2_client_datetime_t) {
+            .tm = *time_info,
+            .msec = 0,
+            .tzname = "America/New_York"
```

Comment:
This is assuming server runs in this timezone, which isn't always right. I'd use `dttz_to_client_datetime`.